### ISSUE TITLE
new debugger: styler issues

### ISSUE THREAD

**Issue by marrub--** [devinacker/bsnes-plus#222]
Not entirely sure how to phrase this, but there are several styling issues with newdebugger that make it quite clumsy to use with my system theme: ![a screen shot of the debugger with somewhat unfitting colors and widget sizes](http://mab.greyserv.net/f/190808-064725.png)

**Comment by devinacker** [comment]
Is there anything particularly problematic you've noticed aside from clashing text/highlight colors now that the font + register widget size stuff got resolved?

**Comment by marrub--** [comment]
Besides the debugger colors, there's also this menu which has hardcoded colors for some reason:

![](http://mab.greyserv.net/f/190810-213938.png)

And the menubar menus which use images for checkboxes and radial buttons, but honestly that one is not necessarily a bad thing (since it looks visually consistent between OSes this way.)

**Comment by devinacker** [comment]
Good catch, that weird path window color thing should be fixed in 15615267e55c801755b34dcd6c36a0204fb6be49

**Comment by devinacker** [comment]
Thinking about this a bit...

It's not really documented right now, but bsnes-plus will automatically apply an application-wide Qt style sheet from the user directory if one exists (at `~/.bsnes/style.qss`. I might just leverage that to better handle stuff that currently uses hardcoded colors by having them use stylesheet classes and then dropping a default `style.qss` in the user directory that can be edited freely.

Not exactly as convenient as having an in-app color picker or anything but it'd still allow customizing the colors of disassembly/trace logs, marked memory regions, etc... What do you think?

**Comment by marrub--** [comment]
Not a bad idea at all, though having a UI would be nice it's certainly not a common issue in the first place.

**Comment by devinacker** [comment]
On second thought, that may just be more of a pain in the ass to implement for the stuff that currently uses QPen/QColor for styling, since that stuff doesn't really have access to stylesheet information... Maybe I should just go the other way around and have them configurable via the GUI and the regular config file instead.
